src/client/v2/algod/algod.ts

```diff
@@ -66,8 +66,8 @@ export default class AlgodClient extends ServiceClient {
       | AlgodTokenHeader
       | CustomTokenHeader
       | BaseHTTPClient,
-    baseServer = 'http://r2.algorand.network',
-    port: string | number = 4180,
+    baseServer: string = '',
```

Comment:
@barnjamin I agree js-algornad-sdk is inconsistent with other SDKs and is arguably inconvenient.  

My concerns are:
* It's a breaking change.  I need a bit of convincing to understand the _bug_ framing.
* If we're making a breaking change, can we instead mark the `baseServer` and `port` as required parameters?  I think I don't see the case for _reasonable_ default values here.
 
Aside:   We're planning a major version bump for Q4 2022:  https://github.com/algorand/js-algorand-sdk/issues/650.  If we land on categorizing the PR as a breaking change, then we can consider adding it to https://github.com/algorand/js-algorand-sdk/issues/650.

Reply:
> I need a bit of convincing to understand the bug framing.

I'm not sure this would break anyone but I have no idea why this port was chosen for default. Maybe it'd help to understand that?

I supposed it's not a bug in the sense that it's broken, more a bug in dev expectations.

> If we're making a breaking change, can we instead mark the baseServer and port as required parameters? I think I don't see the case for reasonable default values here.

I think the port should stay optional since most folks wouldn't consider setting 443 on https hosts and empty string or 0 will allow the client to "figure that out"

Reply:
This would also address the bug reported by #458

Reply:
Circling back after a verbal with @barnjamin and @jasonpaulos.  Here's where we landed:
* Changing default values in a public method _is_ a breaking change. 
* Let's include an improvement in v2.0.0.  Concretely, @barnjamin will make these changes:
  * Change base from `develop` to `integration/2.0.0` (new branch).  `integration/2.0.0` will be the integration branch until v2.0.0 is released.  We can safely merge into it.
  *  Since we'll make a breaking change, let's improve the status quo by removing `port`.  The user can supply port as needed in `baseServer`. 
* Based on these outcomes, I added https://github.com/algorand/js-algorand-sdk/issues/657 to the v2.0.0 epic (https://github.com/algorand/js-algorand-sdk/issues/650).